rfc/src/rfcs/0005-should-fail-attr.md

```diff
@@ -0,0 +1,250 @@
+    let mut device = Device::new();
+    device.init();
+    device.init();
+}
+```
+
+Since we added `#[kani::should_panic]`, running this example would produce a successful code.
+
+Now, we've considered two ways to represent this result in the verification output.
+Note that it's important that we provide the user with this feedback:
+ 1. **(Expectation)** Was Kani expecting the harness to pass or fail?
+ 2. **(Outcome)**: What's the actual result that Kani produced after the analysis?
+This will avoid a potential scenario where the user doesn't know for sure if the attribute has had an effect when verifying the harness.
+
+As mentioned, we've considered two ways to represent this result.
+
+#### Representation #1 (Recommended): No changes to overall result
+
+```rust
+VERIFICATION:- FAILED (expected FAILED)
+```
+
+We recommend this representation so the user receives clear information about both the outcome and the expectation.[^footnote-representation]
+
+#### Representation #2: Changes to overall result
+
+```rust
+VERIFICATION:- SUCCESSFUL (expected FAILED and it FAILED)
+```
+
+### Multiple Harnesses
+
+When there are multiple harnesses, we'll implement the single-harness changes in addition to the following ones.
+Currently, a "Summary" section appears[^footnote-summary] after reporting the results for each harness:
+```
+Verification failed for - harness3
+Verification failed for - harness2
+Verification failed for - harness1
+Complete - 0 successfully verified harnesses, 3 failures, 3 total.
+```
+
+Harnesses marked with `#[kani::should_panic]` won't show unless the expected result was different from the actual result.
+The summary will consider harnesses that match their expectation as "successfully verified harnesses".
+
+Therefore, if we added `#[kani::should_panic]` to all harnesses in the previous example, we'd see this output:
+
+```
+Complete - 3 successfully verified harnesses, 0 failures, 3 total.
+```
+
+### Availability
+
+This feature **will only be available as an attribute**.
+That means this feature won't be available as a CLI option (i.e., `--should-panic`).
+There are good reasons to avoid the CLI option:
+ - It'd make the design and implementation unnecessarily complex.
+ - It'd only be useful when combined with `--harness` to filter negative harnesses.
+ - We could have trouble extending its functionality (see [Future possibilities](#future-possibilities) for more details).
+
+### Pedagogy
+
+The `#[kani::should_panic]` attribute will become one of the most basic attributes in Kani.
+As such, it'll be mentioned in the tutorial and added to the dedicated section planned in [#2208](https://github.com/model-checking/kani/issues/2208).
+
+In general, **we'll also advise against negative verification** when a harness can be written both as a regular (positive) harness and a negative one.
+The feature, as it's presented in this proposal, won't allow checking that the panic failure is due to the panic we expected. 
```

Comment:
This seems like a big deal TBH. From basically the first example, I was wondering "what happens if a new unintentional `panic` later gets inserted earlier in the code?"

Is there a way to 'fix' this? Instead of an attribute, could we have a magic comment (or some Rusty equivalent) directly above the line that is expected to panic, instead of the whole harness?

Reply:
The main problem with the approach you mention (and similar ones like the "function that takes a closure", which is now mentioned as an alternative) is that you can't use them on code that's external to your packages.

Moreover, it seems to me that users just want a coarse mechanism to check that a panic is reachable. It's possible that users want a more fine-grained mechanism, as noted in the RFC, but we simply can't get much data on that unless we have the coarse one first.